pixl_core/src/core/uploader/_sftp.py

```diff
@@ -0,0 +1,148 @@
+#  Copyright (c) University College London Hospitals NHS Foundation Trust
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+"""Uploader subclass for SFTP."""
+
+import os
+from collections.abc import Generator
+from contextlib import contextmanager
+from typing import BinaryIO, Optional
+
+import paramiko
+from loguru import logger
+
+from core.exports import ParquetExport
+from core.uploader._orthanc import StudyTags, get_study_zip_archive
+from core.uploader.base import Uploader
+
+
+class SFTPUploader(Uploader):
+    """Upload strategy for an SFTP server."""
+
+    def __init__(self, project_slug: str, keyvault_alias: Optional[str]) -> None:
+        """Create instance of parent class"""
+        super().__init__(project_slug, keyvault_alias)
+
+    def _set_config(self) -> None:
+        """Set the configuration for the SFTP uploader."""
+        # Use the Azure KV alias as prefix if it exists, otherwise use the project name
+        az_prefix = self.keyvault_alias
+        az_prefix = az_prefix if az_prefix else self.project_slug
+
+        # Get SFTP connection details from keyvault
+        self.host = self.keyvault.fetch_secret(f"{az_prefix}--sftp--host")
+        self.username = self.keyvault.fetch_secret(f"{az_prefix}--sftp--username")
+        self.password = self.keyvault.fetch_secret(f"{az_prefix}--sftp--password")
+        self.port = int(self.keyvault.fetch_secret(f"{az_prefix}--sftp--port"))
+        self.known_hosts_path = self.keyvault.fetch_secret(f"{az_prefix}--sftp--known-hosts-path")
```

Comment:
Might make more sense to pass in the host key itself, instead of a `known_hosts` file. On the other hand, the `known_hosts` file is what you get from the TRE.